### DISCUSSION TITLE
Fetch longer Wikipedia sections or even entire articles

### DISCUSSION THREAD

**Discussion by upward4** [monkeytypegame/monkeytype#2256] [Ideas]
I noticed that Wikipedia mode only fetches the first section of articles. Usually, those sections are quite short as in number of words. It would be nice if Monkeytype could fetch other, longer sections too, or even entire articles at once. As of now, in order to match the number of words set by the user, Monkeytype has to pull from multiple Wikipedia articles at once, resulting in a series of introduction sections of totally different articles.

**Comment by Miodec** [comment]
@Jonny-exe  is this possible?

**Reply by Jonny-exe** [reply]
I haven't found any way of doing it through the Wikipedia API because the only way to get the article in plain text (not in Wikipedia syntax nor HTML) is by getting an extract. You could get it in HTML or Wikipedia syntax and then clean it up with regular expressions, but I don't think it's the best solution.

**Comment by Paulsky** [comment]
I really like the Wikipedia funbox. As an addition to this request, I would like to add a function to require line breaks/enters between the paragraphs or/and different article introductions. Because believe it or not, my enter key is the most unpracticed key. And I would like to practice/use it more.
